### PR TITLE
KdTree: handle 0 or negative k for nearestKSearch

### PR DESCRIPTION
**Pull request by JStech**
While working on #4414 I mistakenly searched for 0 nearest neighbors, and the method segfaulted. This handles that situation better, as well as if a negative k is passed (the argument is a signed integer). Also, my editor automatically removes trailing whitespace, so that's in here too. I could revert all those changes if desired.

### CODE REVIEW

File: kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp [hunk 1]
```diff
@@ -120,39 +120,43 @@ pcl::KdTreeFLANN<PointT, Dist>::setInputCloud (const PointCloudConstPtr &cloud,
...
-template <typename PointT, typename Dist> int 
-pcl::KdTreeFLANN<PointT, Dist>::nearestKSearch (const PointT &point, int k, 
-                                                std::vector<int> &k_indices, 
+template <typename PointT, typename Dist> int
+pcl::KdTreeFLANN<PointT, Dist>::nearestKSearch (const PointT &point, int k,
+                                                std::vector<int> &k_indices,
                                                 std::vector<float> &k_distances) const
 {
   assert (point_representation_->isValid (point) && "Invalid (NaN, Inf) point coordinates given to nearestKSearch!");
+  assert ((k >= 0) && "Invalid number of neighbors");
```
Comment (kunaltyagi) [inline kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp:137]: Why not just accept unsigned int as input? See #4350 (only the changes in octree module)

**Review by kunaltyagi — COMMENTED** [review]
Please split this PR in 2 commits at the very least: 1 with ws changes and one with logic change and one for just logic changes

File: kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp (hunk 1 above)
Comment (JStech) [reply kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp]: Good call. Done.

**Comment by JStech** [comment]
I switched to unsigned ints, and separated the whitespace changes from the actual changes. Let me know if you have any more suggestions.

**Comment by kunaltyagi** [comment]
Why did you choose `unsigned int` over `uindex_t`?

**Comment by mvieth** [comment]
> Why did you choose `unsigned int` over `uindex_t`?

I think that's a good choice because `uindex_t` could create the false impression that `k` is an index, but it is actually a number.

**Comment by mvieth** [comment]
Seems like `total_nr_points_` also needs to be an unsigned data type (to make the macOS clang build pass)

**Comment by JStech** [comment]
Yes, I chose `unsigned int` because `k` is not an index, and I don't think we should plan for it to be arbitrarily large (4 billion nearest neighbors ought to be enough for anybody).

I did make `total_nr_points_` a `uindex_t`, and I was able to eliminate several casts as a result. I'll let CI tell me if I messed it up.

**Comment by JStech** [comment]
I think this is ready to go. It passed CI.

**Review by kunaltyagi — APPROVED** [review]

**Review by mvieth — APPROVED** [review]
It would probably make sense to change `k` from `int` to `unsigned int` in all `nearestKSearch` functions (search module) in the future.

**Comment by huonggiangdhtn** [comment]
`Neighbors within radius search at (0.144053 0.15847 0.434392) with radius=0.01
    0.141197 0.161906 0.436549 (squared distance: 2.46188e-05)
    0.143541 0.163574 0.433012 (squared distance: 2.82178e-05)
    0.0257284 -0.115956 0.444 (squared distance: 6.19848e-05)`

I use this function to find nearest point with radius, kdtree.radiusSearch (searchPoint, radius, pointIdxRadiusSearch, pointRadiusSquaredDistance). However this function return error with one of these coordinate < 0 like the third row in above example. Can anyone know how to fix it

**Pull request merged**